src/main/java/frc/robot/util/Vision.java

```diff
@@ -0,0 +1,176 @@
+      //   new Pair<PhotonCamera, Transform3d>(camera2, VisionConstants.kCameraToRobot2)
+      // ));
+      // robotPoseEstimator2 = new RobotPoseEstimator(aprilTagFieldLayout, PoseStrategy.AVERAGE_BEST_TARGETS, camList);    
+    }
+
+  }
+
+  public static AprilTagFieldLayout getTagFieldLayout() {
+    // AprilTagFields.loadAprilTagLayoutField();
+    try {
+      //TODO: Fix this
+      // aprilTagFieldLayout = AprilTagFieldLayout.loadFromResource(AprilTagFields.k2023ChargedUp.m_resourceFile);
+      aprilTagFieldLayout = AprilTagFieldLayout.loadFromResource(AprilTagFields.k2022RapidReact.m_resourceFile);
+    } catch (IOException ex) {
+      aprilTagFieldLayout = new AprilTagFieldLayout(VisionConstants.kTagPoses, FieldConstants.kFieldLength, FieldConstants.kFieldWidth);
+      System.out.println("Vision setup IOException: "+ex.getMessage());
+    }
+
+    return aprilTagFieldLayout;
+  }
+
+  // public Map<Pose3d, Double> getPoseEstimation() {
+  //   if (m_hasTargets) {
+  //     double imageCaptureTime = Timer.getFPGATimestamp() - m_latency;
+  //     Transform3d camToTargetTrans = m_bestTarget.getBestCameraToTarget();
+  //     Pose3d camPose = VisionConstants.kTagPoses.get(m_targetId).transformBy(camToTargetTrans.inverse());
+  //     return Collections.singletonMap(camPose.transformBy(VisionConstants.kCameraToRobot), imageCaptureTime);
+  //   }
+  //   return null;
+  // }
+
+  /**
+   * Prints the estimated pose from the vision
+   */
+  public static void printEstimate(){
+    robotPoseEstimator.setReferencePose(new Pose3d(0, 0, 0, new Rotation3d(0, 0, 0)));
+    Optional<Pair<Pose3d,Double>> r = robotPoseEstimator.update();
+    if(r.isPresent()){
+      System.out.printf("Present: %b\nPose: (%.2f, %.2f, %.2f)\nRotation: %.2f degrees\nTime: %.2f\n",
+        r.isPresent(), 
+        r.get().getFirst().getX(), 
+        r.get().getFirst().getY(), 
+        r.get().getFirst().getZ(), 
+        r.get().getFirst().getRotation().toRotation2d().getDegrees(), 
+        r.get().getSecond()
+      );
+    }else{
+      System.out.println("Result is not present");
+    }
+  }
+
+  public static Optional<Pair<Pose3d,Double>> getEstimatedGlobalPose(Pose2d prevEstimatedRobotPose) {
+    robotPoseEstimator.setReferencePose(prevEstimatedRobotPose);
+    Optional<Pair<Pose3d,Double>> r = robotPoseEstimator.update();
+    return r;
+
+    // double currentTime = Timer.getFPGATimestamp();
+    // Optional<Pair<Pose3d, Double>> result = m_robotPoseEstimator.update();
+    // if (result.isPresent()) {
+    //     return new Pair<Pose2d, Double>(result.get().getFirst().toPose2d(), currentTime - result.get().getSecond());
+    // } else {
+    //     return new Pair<Pose2d, Double>(null, 0.0);
+    // }
+}
+
+/**
+ * Not used
+ * @param prevEstimatedRobotPose Doesn't do anything
+ * @return null
+ */
+public static Optional<Pair<Pose3d,Double>> getEstimatedGlobalPose2(Pose2d prevEstimatedRobotPose) {
```

Comment:
consider removing this

Reply:
This is for a version that doesn't work. If we decide to go back to it, I don't want to rewrite it.